proteuslib/unit_models/tests/test_reverse_osmosis_0D.py

```diff
@@ -649,23 +655,37 @@ def test_Pdrop_calculation(self):
 
         # test solution
-        assert (pytest.approx(-9.173e5, rel=1e-3) == value(m.fs.unit.deltaP[0]))
-        assert (pytest.approx(1.904e-3, rel=1e-3) ==
+        assert (pytest.approx(151.622, rel=1e-3) == value(m.fs.unit.N_Re_io[0, 'in']))
+        assert (pytest.approx(0.2089, rel=1e-3) == value(m.fs.unit.velocity_io[0, 'in']))
+        assert (pytest.approx(63.001, rel=1e-3) == value(m.fs.unit.N_Re_io[0, 'out']))
+        assert (pytest.approx(0.0914, rel=1e-3) == value(m.fs.unit.velocity_io[0, 'out']))
+        assert (pytest.approx(151.622, rel=1e-3) == value(m.fs.unit.velocity_io[0, 'in']
+                * m.fs.unit.feed_side.properties_in[0].dens_mass_phase['Liq']
+                * m.fs.unit.dh
+                / m.fs.unit.feed_side.properties_in[0].visc_d_phase['Liq']))
+        assert (pytest.approx(63.001, rel=1e-3) == value(m.fs.unit.velocity_io[0, 'out']
+                * m.fs.unit.feed_side.properties_out[0].dens_mass_phase['Liq']
+                * m.fs.unit.dh
+                / m.fs.unit.feed_side.properties_out[0].visc_d_phase['Liq']))
+        assert (pytest.approx(-2.786e5, rel=1e-3) == value(m.fs.unit.deltaP[0]))
+        assert (pytest.approx(-0.3483e5, rel=1e-3) == value(m.fs.unit.deltaP[0]/m.fs.unit.length))
+        assert (pytest.approx(1.097e-2, rel=1e-3) ==
                 value(m.fs.unit.flux_mass_phase_comp_avg[0, 'Liq', 'H2O']))
-        assert (pytest.approx(1.727e-6, rel=1e-3) ==
+        assert (pytest.approx(2.389e-6, rel=1e-3) ==
                 value(m.fs.unit.flux_mass_phase_comp_avg[0, 'Liq', 'NaCl']))
-        assert (pytest.approx(0.0952, rel=1e-3) ==
+        assert (pytest.approx(0.5485, rel=1e-3) ==
                 value(m.fs.unit.properties_permeate[0].flow_mass_phase_comp['Liq', 'H2O']))
-        assert (pytest.approx(8.637e-5, rel=1e-3) ==
+        assert (pytest.approx(1.194e-4, rel=1e-3) ==
                 value(m.fs.unit.properties_permeate[0].flow_mass_phase_comp['Liq', 'NaCl']))
         assert (pytest.approx(35.751, rel=1e-3) ==
                 value(m.fs.unit.feed_side.properties_in[0].conc_mass_phase_comp['Liq', 'NaCl']))
-        assert (pytest.approx(53.561, rel=1e-3) ==
+        assert (pytest.approx(49.2250, rel=1e-3) ==
                 value(m.fs.unit.feed_side.properties_interface_in[0].conc_mass_phase_comp['Liq', 'NaCl']))
-        assert (pytest.approx(39.524, rel=1e-3) ==
+        assert (pytest.approx(81.4165, rel=1e-3) ==
                 value(m.fs.unit.feed_side.properties_out[0].conc_mass_phase_comp['Liq', 'NaCl']))
-        assert (pytest.approx(46.958, rel=1e-3) ==  # TODO: expected this value to be higher than interface concentration at inlet, but bypassing for now- has to do with pressure drop
+        assert (pytest.approx(87.7047, rel=1e-3) ==
                 value(m.fs.unit.feed_side.properties_interface_out[0].conc_mass_phase_comp['Liq', 'NaCl']))
+        assert value(m.fs.unit.df) <= value(m.fs.unit.channel_height)
```

Comment:
Usually we don't check user inputs

Reply:
This was really to check that the constraint is in place.

Reply:
It doesn't check if the constraint is in place, these values are fixed in the test case. An isinstance(constraint_name, Constraint) would check if it exists, but as mentioned above I think we should drop the constraint anyways.